dra-evolution/pkg/api/capacity_types.go

```diff
@@ -59,14 +59,11 @@ type ResourcePoolSpec struct {
 	// vendor of the driver.
 	DriverName string `json:"driverName" protobuf:"bytes,3,name=driverName"`
 
-	// SharedCapacity defines the set of shared capacity consumable by
-	// devices in this ResourceSlice.
+	// CommonData represents information that is common across various devices,
+	// to reduce redundancy in the devices list.
 	//
-	// Must not have more than 128 entries.
-	//
-	// +listType=atomic
 	// +optional
-	SharedCapacity []SharedCapacity `json:"sharedCapacity,omitempty"`
+	CommonData []CommonDeviceData `json:"commonData,omitempty"`
```

Comment:
I wouldn't include this. If we are ripping out the model that supports partitioning, I think we should should just go back to an explicit list of attributes per device.

Reply:
We need an extension point to insert the things we are debating in #20. That's the primary reason that is here. Adding the common things was just a way to make it have some meaning and not be a stub. With this, an old scheduler an see that there is an entry in this list, but when it unmarshals it, all the fields the scheduler is aware of are nil, so it knows "ooo, I should ignore this slice, I don't know what to do with it".

If we don't have this, then we need some other way for an older scheduler to know that "it doesn't understand this object".

Reply:
My current thinking is that we would *not* have any extension points directly in the device at this point. Instead every device will  have the ability to point to one (and exactly one) DeviceShape/DeviceTemplate. That is what will have all of these extra bells and whistles. The device itself is still just a named list of attributes (and now capacities).

Reply:
Ok, that works for me. The thing is, we still need a way for old schedulers to know "this slice is not something I understand". So a "SliceType" discriminator perhaps? Or we have to have a field like I have done here, that has a one-of.